tests/functional/z_cluster/nodes/test_nodes_restart_hci.py

```diff
@@ -378,3 +379,33 @@ def test_stop_provider_mgr_node_and_check_client_connection(
         logger.info(f"Starting the MGR node {mgr_node.name}")
         nodes.start_nodes(nodes=[mgr_node], wait=True)
         self.sanity_helpers.health_check_provider_mode(tries=40)
+
+    @tier4a
+    @pytest.mark.parametrize(
+        "cluster_type",
+        [HCI_CLIENT],
+    )
+    def test_hosted_cluster_nodes_restart(self, cluster_type, nodes):
```

Comment:
This is something I took from the Managed Service. The `cluster_type` parameter is used for the `setup` method. We use it here https://github.com/red-hat-storage/ocs-ci/blob/master/tests/functional/z_cluster/nodes/test_nodes_restart_hci.py#L63: `switch_to_correct_cluster_at_setup(request)`. This will switch to the correct cluster type: `provider` or `client` in this case.